pybamm/experiments/experiment.py

```diff
@@ -0,0 +1,220 @@
+    Discharge at C/20 for 0.5 hours,
+    Charge at 0.5 C for 45 minutes,
+    Discharge at 1 A for 90 seconds,
+    Charge at 200 mA for 45 minutes (1 minute period),
+    Discharge at 1 W for 0.5 hours,
+    Charge at 200 mW for 45 minutes,
+    Rest for 10 minutes (5 minute period),
+    Hold at 1 V for 20 seconds,
+    Charge at 1 C until 4.1 V,
+    Hold at 4.1 V until 50 mA,
+    Hold at 3 V until C/50,
+    """
+
+
+class Experiment:
+    """
+    Base class for experimental conditions under which to run the model
+
+    Parameters
+    ----------
+    operating_conditions : list
+        List of operating conditions
+    parameters : dict
+        Dictionary of parameters to use for this experiment, replacing default
+        parameters as appropriate
+    period : string, optional
+        Period (1/frequency) at which to record outputs. Default is 1 minute. Can be
+        overwritten by individual operating conditions.
+
+    """
+
+    def __init__(self, operating_conditions, parameters=None, period="1 minute"):
+        self.period = self.convert_time_to_seconds(period.split())
+        self.operating_conditions_strings = operating_conditions
+        self.operating_conditions, self.events = self.read_operating_conditions(
+            operating_conditions
+        )
+        parameters = parameters or {}
+        if isinstance(parameters, dict):
+            self.parameters = parameters
+        else:
+            raise TypeError("experimental parameters should be a dictionary")
+
+    def __str__(self):
+        return str(self.operating_conditions_strings)
+
+    def __repr__(self):
+        return "pybamm.Experiment({!s})".format(self)
+
+    def read_operating_conditions(self, operating_conditions):
+        """
+        Convert operating conditions to the appropriate format
+
+        Parameters
+        ----------
+        operating_conditions : list
+            List of operating conditions
+
+        Returns
+        -------
+        operating_conditions : list
+            Operating conditions in the tuple format
+        """
+        converted_operating_conditions = []
+        events = []
+        for cond in operating_conditions:
+            if isinstance(cond, str):
+                next_op, next_event = self.read_string(cond)
+                converted_operating_conditions.append(next_op)
+                events.append(next_event)
+            else:
+                raise TypeError(
+                    """Operating conditions should be strings, not {}. For example: {}
+                    """.format(
+                        type(cond), examples
+                    )
+                )
+
+        return converted_operating_conditions, events
+
+    def read_string(self, cond):
+        """
+        Convert a string to a tuple of the right format
+
+        Parameters
+        ----------
+        cond : str
+            String of appropriate form for example "Charge at x C for y hours". x and y
```

Comment:
It might be useful to have the structure of the strings explained in the docs for experiments (or somewhere else more prominent)

Reply:
have expanded the docstrings. it prints out a load of examples if it doesn't recognize the string, so hopefully will be intuitive enough